osquery/killswitch/killswitch_plugin.cpp

```diff
@@ -0,0 +1,89 @@
+#include <string>
+
+#include <chrono>
+#include <osquery/flags.h>
+#include <osquery/killswitch.h>
+#include <osquery/killswitch/killswitch_plugin.h>
+#include <osquery/registry_factory.h>
+
+namespace osquery {
+
+CREATE_REGISTRY(KillswitchPlugin, "killswitch");
+
+Expected<std::map<std::string, bool>, KillswitchPlugin::ParseMapJSONError>
+KillswitchPlugin::parseMapJSON(const std::string& content) {
+  std::map<std::string, bool> result;
+
+  auto doc = JSON::newObject();
+  if (!doc.fromString(content) || !doc.doc().IsObject()) {
+    return createError(
+        KillswitchPlugin::ParseMapJSONError::UnknownParsingProblem,
+        "Error parsing the killswitch JSON. Content : " + content);
+  }
+
+  for (const auto& keyValue : doc.doc().GetObject()) {
+    if (!keyValue.name.IsString()) {
+      return createError(KillswitchPlugin::ParseMapJSONError::IncorrectKeyType,
```

Comment:
My point is that killswitch or its config is not supposed to be the place where errors are okay and at least part of it will work.
That way it would be harder to ignore the problem and it will be resolved soon. But I'm open to your thoughts and best practices ofc.   @muffins